pkg/crd/gen_integration_test.go

```diff
@@ -119,6 +119,32 @@ var _ = Describe("CRD Generation proper defaulting", func() {
 		expectedOut := string(expectedFileFoos) + string(expectedFileZoos)
 		Expect(out.buf.String()).To(Equal(expectedOut), cmp.Diff(out.buf.String(), expectedOut))
 	})
+
+	It("should add preserveUnknownFields=false when specified", func() {
```

Comment:
Let's also add a case that it doesn't show up with false if PreserveUnknownFields is nil

Reply:
Added